client/src/redux/prop-types.ts

```diff
@@ -310,6 +317,23 @@ export type ClaimedCertifications = {
   isMachineLearningPyCertV7: boolean;
 };
 
+export type SavedChallenges = SavedChallenge[];
+
+export type SavedChallenge = {
+  id: string;
+  challengeFiles: SavedChallengeFiles;
+};
+
+export type SavedChallengeFile = {
+  fileKey: string;
+  ext: Ext;
+  name: string;
+  history?: string[];
+  contents: string;
+};
+
+export type SavedChallengeFiles = SavedChallengeFile[] | null;
```

Comment:
Seeing as we have control over this, do we want this to be initialised as `[]` or `null`?